### PR TITLE
Use custom user agent string at all times

### PR DESCRIPTION
**Pull request by laurenyu**
The previous implementation allowed for our user agent string to be overridden if the user supplied a boto3 session, SageMaker client, or SageMaker runtime client.

### CODE REVIEW

File: src/sagemaker/session.py [hunk 1]
```diff
@@ -17,20 +17,24 @@
...
 import six
 from botocore.exceptions import ClientError
 
 from sagemaker.utils import name_from_image
 import sagemaker.logs
 
-SDK_VERSION = pkg_resources.require('sagemaker')[0].version
 
+SDK_VERSION = pkg_resources.require('sagemaker')[0].version
+OS_NAME = platform.system()
```
Comment (owenataws) [inline src/sagemaker/session.py]: If no OS name can be resolved, ``platform.system()`` will return an empty string. Can this be something like:

``OS_NAME = platform.system() or 'UnresolvedOS'``

Same for ``OS_VERSION``

File: src/sagemaker/session.py [hunk 2]
```diff
@@ -71,30 +75,20 @@ def __init__(self, boto_session=None, sagemaker_client=None, sagemaker_runtime_c
...
+        user_agent_string = 'AWS-SageMaker-Python-SDK/{} Python/{} {}/{} Boto3/{} Botocore/{}'\
+            .format(SDK_VERSION, PYTHON_VERSION, OS_NAME, OS_VERSION, boto3.__version__, botocore.__version__)
 
-        # Some information is appended automatically, so this creates a user agent string that looks like
-        # 'AWS-SageMaker-Python-SDK/1.0/1.7.45 Python/3.6.2 Darwin/15.6.0 Boto3/1.4.7 Botocore/1.7.45'
-        botocore_session.user_agent_name = 'AWS-SageMaker-Python-SDK/{}'.format(SDK_VERSION)
-        botocore_session.user_agent_extra = 'Boto3/{} Botocore/{}'.format(boto3.__version__,
-                                                                          botocore_session.user_agent_version)
+        self.sagemaker_client = sagemaker_client or self.boto_session.client('sagemaker')
+        self.sagemaker_client._client_config.user_agent = user_agent_string
```
Comment (owenataws) [inline src/sagemaker/session.py]: Is there a reason we don't preserve the original user agent string as well. Why not just append the SageMaker string to the end?

Comment (laurenyu) [reply src/sagemaker/session.py]: The default user agent string is 'Boto3/1.4.8 Python/3.6.3 Darwin/15.6.0 Botocore/1.8.6', which is why I just added the Boto and Botocore versions to the end. It needs to start with 'AWS-SageMaker' and be followed by 'language/version OS/version' for parsing. Appending the original user agent string here is a good call, though.

File: src/sagemaker/session.py (hunk 1 above)
Comment (laurenyu) [reply src/sagemaker/session.py]: done. Also moved everything to a new module.

**Pull request merged**